pkg/cmd/util/route.go

```diff
@@ -14,7 +14,10 @@ import (
 // UnsecuredRoute will return a route with enough info so that it can direct traffic to
 // the service provided by --service. Callers of this helper are responsible for providing
 // tls configuration, path, and the hostname of the route.
-func UnsecuredRoute(kc kclientset.Interface, namespace, routeName, serviceName, portString string) (*api.Route, error) {
+// useTargetPort choses the service TargetPort instead of Port for the route TargetPort.
```

Comment:
Perhaps we should change useTargetPort to forcePort

And then we can say "forcePort always sets a port, even when there is only one and it has no name."